src/System.IO.FileSystem/src/System/IO/UnixFileSystem.cs

```diff
@@ -653,5 +653,10 @@ public override IFileSystemObject GetFileSystemInfo(string fullPath, bool asDire
                 (IFileSystemObject)new DirectoryInfo(fullPath, null) :
                 (IFileSystemObject)new FileInfo(fullPath, null);
         }
+
+        public override string[] GetLogicalDrives()
+        {
+            return DriveInfoInternal.GetLogicalDrives();
```

Comment:
> there is still the view from common- it's confusing to see just one platform specific file there

There are lots of platform-specific files already in Common and that don't have counterparts for other platforms, e.g. the PersistedFiles files in Common/src/System/IO, RowConfigReader in Common/src/System/IO. all of the files under Common/src/System/Net/Security, etc., nevermind all of the files under each platform's Interop folder, which are all by-definition platform-specific and without counterparts.  The "view from Common" is not meant to be one where every platform-specific file has a counterpart; it's simply an organized location of files that are shared between assemblies, nothing more, nothing less.

> There also isn't a guarantee on how many consumers there will be in the future

The number of consumers will go down in the future, not up.  There should really only be at most one assembly in corefx with this functionality compiled into it.  Once Environment.GetLogicalDrives() is implemented and exposed in a contract, both System.IO.FileSystem.dll and System.IO.FileSystem.DriveInfo.dll can and should be implemented in terms of that, rather than recompiling this functionality into each assembly.  So the functionality you're adding to Common will likely just be deleted, and the more files you add, the more files are going to need to be cleaned up and deleted.